src/hosted-form/hosted-form.ts

```diff
@@ -21,7 +22,8 @@ export default class HostedForm {
         private _eventListener: IframeEventListener<HostedInputEventMap>,
         private _payloadTransformer: HostedFormOrderDataTransformer,
         private _eventCallbacks: HostedFormEventCallbacks,
-        private _paymentHumanVerificationHandler: PaymentHumanVerificationHandler
+        private _paymentHumanVerificationHandler: PaymentHumanVerificationHandler,
+        ppsdkStepHandler?: StepHandler
```

Comment:
Could we make this a required dependency rather than optional? And instead of conditionally handling `HostedInputEventType.SubmitSucceeded` event depending on whether `StepHandler` is injected or not, what we can do is have the event carry a flag to indicate what kind of `payload` it is carrying (i.e.: if it's PPSDK response, handle it with `StepHandler`). That way it is more explicit and the same `HostedForm` instance is compatible with both non-PPSDK and PPSDK payment methods.

Reply:
This means that we will inject the `StepHandler` into other strategies that create a hosted form while we know that they don't need the `StepHandler`, like `StripeV3PaymentStrategy`, `MonerisPaymentStrategy` as well as the generic `CreditCardPaymentStrategy` and `CreditCardRedirectPaymentStrategy`. Why would we do this?

Reply:
You wouldn't though because the construction of `HostedForm` is handled by `HostedFormFactory`, so all these strategies don't need to supply the dependencies required to construct `HostedForm` or know about `StepHandler`. For example, you'd supply `StepHandler` in a similar way as `PaymentHumanVerificationHandler` inside the factory method.

Reply:
https://github.com/bigcommerce/checkout-sdk-js/pull/1396/files#r848936962

Reply:
I get what you mean, but please check [my comment below](/bigcommerce/checkout-sdk-js/pull/1396#discussion_r848936962) about the issue.